File: src/main/resources/static/app/services/postService.js
angular.module('iBlog').factory('postService', function ($resource, $q, $http) {
    var postService = {}

    postService.getAllPosts = function () {
        var defer = $q.defer();
        $http.get('/posts').
        then(function (response) {
            defer.resolve(response.data);
        }).catch(defer.reject);
        return defer.promise
    }
    postService.getPost = function (postId) {
        var defer = $q.defer();
        $http.get('/posts/' + postId).
        then(function (response) {
            defer.resolve(response.data);
        }).catch(defer.reject);
        return defer.promise
    }



    postService.addComment = function (comment) {
        var defer = $q.defer();
        $http.post('/posts/comments', comment).
        then(function (response) {
            defer.resolve(response.data);
        }).catch(defer.reject);
        return defer.promise
    }



    postService.addPost = function (newpost) {
        var defer = $q.defer();
        $http.post('/posts', newpost).
        then(function (response) {
            defer.resolve(response.data);
        }).catch(defer.reject);
        return defer.promise
    }

    //    postService.addLikes = function (postId) {
    //
    //        var post = this.getPost(postId);
    //        //        post.likes = post.likes++;
    //        //        posts.push(postId);
    //        return;
    //    }

    return postService;
});

File: src/main/java/com/easyblogs/services/PostService.java
package com.easyblogs.services;

import java.util.ArrayList;
import java.util.List;
import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.easyblogs.model.Comment;
import com.easyblogs.model.Post;
import com.easyblogs.repository.CommentRepository;
import com.easyblogs.repository.PostRepository;



@Service
public class PostService {
	@Autowired
	PostRepository postRepository;
	
	@Autowired
	CommentRepository commentRepository;

	public List<Post> getAllPosts() {
		ArrayList<Post> posts = new ArrayList<>();
		this.postRepository.findAll().forEach(posts::add);
        return posts;
	}

	public Post addPost(Post post) {
		 return this.postRepository.save(post);
	}
	
	public List<Comment> getAllComments(Long postId)
	{
		ArrayList<Comment> comments = new ArrayList<>();
		//this.commentRepository.findByPostId(postId).forEach(comments::add);
        return comments;
	}

	public Optional<Post> findByPostId(Long postId) {
		return this.postRepository.findById(postId);
	}

	public void addComment(Comment comment) {
		this.commentRepository.save(comment);
	}

}
